internal/logrus_test.go

```diff
@@ -85,3 +85,27 @@ func TestLogrusLoggerInvalidConfig(t *testing.T) {
 
 	NewLogrusLogger("invalid level", nil)
 }
+
+func TestFirstStackTracerInErrorChainWithOneError(t *testing.T) {
+	rootError := errors.New("root error")
+
+	_, ok := rootError.(stackTracer)
+	assert.True(t, ok)
+
+	gotError := firstStackTracerInErrorChain(rootError)
+
+	assert.Equal(t, "root error", gotError.Error())
+}
+
+func TestFirstStackTracerInErrorChainWithMultipleErrors(t *testing.T) {
+	rootError := errors.New("root error")
+	stackTracerError := errors.Wrap(rootError, "wrap root error")
+	newContextError := errors.WithMessage(stackTracerError, "new context added")
+
+	_, ok := newContextError.(stackTracer)
+	assert.False(t, ok)
+
+	gotError := firstStackTracerInErrorChain(newContextError)
+
+	assert.Equal(t, "wrap root error: root error", gotError.Error())
```

Comment:
You shouldn't be testing the error message but the type cast to stackTracer.
```golang
_, ok := gotError.(stackTracer)
assert.True(t, ok)
```

Reply:
I see your point. I think both assertions are important actually. Use `assert.EqualError` though.
```golang
assert.EqualError(t, gotError, stackTracerError.Error())

_, ok := gotError.(stackTracer)
assert.True(t, ok)
```